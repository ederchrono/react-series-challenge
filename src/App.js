import React, { Component } from 'react';
import { BrowserRouter, Route } from 'react-router-dom'
import { connect } from 'react-redux'

import Navbar from './Navbar'
import Home from './Home'
import Favorites from './Favorites'
import Details from './Details'
import './App.css';

class App extends Component {
  render () {
    return (
      <BrowserRouter>
        <div className="App">
          <Navbar />
          <p className="App-intro">
            To get started, edit <code>src/App.js</code> and save to reload.
          </p>
          <Route exact path="/" component={Home}/>
          <Route path="/favs" component={Favorites} />
          <Route path="/details/:id" component={Details} />

          <p>{this.props.count}</p>
          <button onClick={this.props.increment}>+</button>
          <button  onClick={this.props.decrement}>-</button>
        </div>
      </BrowserRouter>
    );
  }
}

function mapStateToProps(state) {
  return {
    count: state
  };
}

function mapDispatchToProps (dispatch) {
  return {
    increment: () => dispatch({ type: 'INCREMENT', payload: {amount: 2} }),
    decrement: () => dispatch({ type: 'DECREMENT' })
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
