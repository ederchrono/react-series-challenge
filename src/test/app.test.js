import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';
import gifData from './data/gifTestData'
import configureStore from 'redux-mock-store'
import { shallowToJson } from 'enzyme-to-json';

const mockStore = configureStore()
const initialState = {
  favoriteGifs: [
    { id: gifData.id }
  ]
}

describe('App', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<App store={mockStore(initialState)} />)
    // expect(wrapper).toMatchSnapshot();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
})
