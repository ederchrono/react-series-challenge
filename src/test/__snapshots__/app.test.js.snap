// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<App
  fetchFavorites={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
